### PR TITLE
Add docstrings for Score() object

### PR DESCRIPTION
**Pull request by tylercroberts**
Changed most of the methods to be hidden. 

I figured we can either automatically compute the score at initialization (treating it nearly as a function) if x and y data are provided immediately, or setup a generic Score object for a particular model and score metrics, and then provide data afterwards in the `Score.calculate()` method.

Filled out function arguments and docstrings for everything in this object as well.

### CODE REVIEW

**Comment by tylercroberts** [comment]
Added an overwrite for Score.__str__() as well, but commented it out so that we still have *a* print method until the code has actually been written.

File: ezmodel/ezmodel.py [hunk 1]
```diff
@@ -1,53 +1,87 @@
...
+    def __init__(self, model, score_type='mse', x=None, y=None):
+        """
+        Constructor for score object. Adds in the model as well as the score type you are looking for.
+        # Could score_type be a list?
+
+        Optionally, you could add in x and y values here, to immediately compute a score.
+
+
+        Args:
+            model (sklearn object): Previously initialized sklearn classifier or regression object with fit & predict methods.
```
Comment (conleyst) [inline ezmodel/ezmodel.py]: I think that we should specify that the model object we pass in isn't trained.

File: ezmodel/ezmodel.py [hunk 2]
```diff
@@ -1,53 +1,87 @@
...
 
-def _coerce(x):
-    """Utility function to coerce data into the correct types to be passed to sklearn"""
-    pass
-
-
-
+        Args:
+            x (ndarray): (d x n) array of features.
+            y (ndarray): (1 x n) array of labels.
```
Comment (conleyst) [inline ezmodel/ezmodel.py]: Swap dimensions

File: ezmodel/ezmodel.py [hunk 3]
```diff
@@ -1,53 +1,87 @@
...
+        Optionally, you could add in x and y values here, to immediately compute a score.
+
+
+        Args:
+            model (sklearn object): Previously initialized sklearn classifier or regression object with fit & predict methods.
+                                    Can also be a pipeline with several steps.
+
+            score_type (list or str): Should be one of: [mse, r2, adj_r2, auc, ...].
+                                      If a list, then a list containing several of those entries as elements.
+            x (ndarray): (d x n) array of features.
```
Comment (conleyst) [inline ezmodel/ezmodel.py]: Swap dimensions

**Review by conleyst — CHANGES_REQUESTED** [review]

**Review by conleyst — APPROVED** [review]

**Pull request merged**